### ISSUE TITLE
Browser JS - Tests sometimes get executed twice

### ISSUE THREAD

**Issue by JoonasC** [kotest/kotest#2723]
**Which version of Kotest are you using**
5.0.1

I have specs with property tests, and sometimes the tests will be executed twice. This only happens with property tests.

It could also have something to do with the fact that the tests take a long time to run.

![image](https://user-images.githubusercontent.com/3898724/145436635-dca4b3fa-7996-43cd-a87c-d4b2884e626a.png)

The tests and general setup is similar to the reproducer from #2697 (with `useMocha()` in the gradle build file).

**Comment by sksamuel** [comment]
It's not an error with output as the times are slightly different, although the `the syntax errors in malformed input should be detected` is the exact same time. Can you re-run this a few times to get some more timing info ?

**Comment by JoonasC** [comment]
I've re-run it a few times, and the timings keep varying. Also the intellij idea test window shows the tests executing, when they are being executed twice. So this is not a bug with the output.

**Comment by sksamuel** [comment]
Are all of those tests data tests or is `the syntax errors in malformed input should be detected` something else ?

**Comment by JoonasC** [comment]
Yes, they are all data/property tests although some of them include standard unit tests.

For example:
```kotlin
"test" {
    "a" shouldBe "a"
    checkAll<Double> { double ->
        ...
    }
}
```

**Comment by sksamuel** [comment]
Kotest has different semantics for data vs property tests. So just to be clear on this error, you're talking about property tests - nesting checkAll inside a "test" block ?

**Comment by JoonasC** [comment]
Some of the tests have nesting `chekAll` functions and some don't, but all of them use the `checkAll` function.

**Comment by sksamuel** [comment]
Thanks, will try to reproduce.

**Comment by JoonasC** [comment]
As I mentionned in the first comment, the reproducer from #2697 is quite similar to my setup (just change `useKarma` to `useMocha`). You can try copying the property test I had in that reproducer a few times in the spec, maybe change it a bit and I think that should reproduce the issue.

**Comment by JoonasC** [comment]
One important detail is that if the `build` directory doesn't exist yet (as in the tests haven't been executed at all yet), this issue doesn't appear. It only appears after the tests have already been executed a few times previously.

**Comment by sksamuel** [comment]
Added these and can't seem to duplicate
https://github.com/kotest/kotest-examples-javascript/blob/master/src/jsTest/kotlin/io/kotest/examples/js/PropertyTests.kt

What happens if you run those ?

**Comment by JoonasC** [comment]
@sksamuel Actually, I just noticed that I'm still using the `5.0.0` gradle multiplatform plugin version. Does that have newer versions available?

**Comment by JoonasC** [comment]
Could this have been caused by that?

**Comment by sksamuel** [comment]
I don't think so. That hasn't changed.

**Comment by sksamuel** [comment]
You can try 5.0.2 to see ?

**Comment by JoonasC** [comment]
@sksamuel Ok, I've updated Kotest and haven't seen this behavior in a while.
Should I close this?

**Comment by sksamuel** [comment]
Yep, it sounds like it was the multi platform race condition fixed in 5.0.2 or 5.0.3.
